registry/auth.go

```diff
@@ -74,6 +74,12 @@ func (auth *RequestAuthorization) getToken() (string, error) {
 	return "", nil
 }
 
+// Checks that requests to the v2 registry can be authorized.
+func (auth *RequestAuthorization) IsAuthorized() bool {
```

Comment:
This is only checking the authentication credentials with the authorization server. The authorization to the registry may still be rejected because either the proper scope was not granted by the authorization server, or the basic auth directly with the registry failed.